kernel.c

```diff
@@ -372,10 +370,9 @@ int k_modoru_get_factory_firmware(void) {
 
   unsigned int factory_fw = -1;
 
-  void *sysroot = ksceKernelGetSysrootBuffer();
-  if (sysroot) {
-    factory_fw = *(unsigned int *)(sysroot + 8);
-	if (*(unsigned int *)(sysroot + 4) > 0x03700011)
+  int res = ksceSblAimgrGetSMI(&factory_fw);
+  if (res >= 0) {
+	if (factory_fw > 0x03700011)
```

Comment:
kblparam+4 is current fw, kblparam+8 is minfw.
modoru checks current fw and uses additional patches if its above 3.70